### PR TITLE
DrawFov fixed version

### PR DESCRIPTION
**Pull request by newcommerdontblame**
shows %90 correct
aimlock=green circle
silent= red circle

### CODE REVIEW

**Comment by g1gantino** [comment]
Works good, thanks for sharing.

File: Osiris/Hacks/Misc.cpp [hunk 1]
```diff
@@ -412,6 +426,24 @@ void Misc::drawBombTimer() noexcept
...
+    if (config->misc.drawFov)
+    {
+        if (!local || !local->isAlive()) return;
+        int weaponId = getWeaponIndex(localPlayer->getActiveWeapon()->itemDefinitionIndex2());
+        if (!config->aimbot[weaponId].enabled) weaponId = 0;
+        if (!config->aimbot[weaponId].enabled) return;
+        auto screenSize = interfaces->surface->getScreenSize();
+        if (config->aimbot[weaponId].silent) interfaces->surface->setDrawColor(255, 10, 10, 255);
+        else interfaces->surface->setDrawColor(10, 255, 10, 255);
+        float radius = std::tan(degreesToRadians(config->aimbot[weaponId].fov) / 2.f) / std::tan(degreesToRadians(local->isScoped() ? local->fov() : (105 + config->visuals.fov)) / 2.f) * screenSize.first;
```
Comment (Kron1Q) [inline Osiris/Hacks/Misc.cpp:441]: float radius = std::tan(degreesToRadians(config->aimbot[weaponIndex].fov) / 2.f) / std::tan(degreesToRadians(globals::currentFOV) / 2.f) * width;

globals::currentFOV = setup->fov;

With your method drawfov wont be accurate with different FOV.

File: Osiris/Hacks/Misc.cpp [hunk 2]
```diff
@@ -412,6 +426,24 @@ void Misc::drawBombTimer() noexcept
     }
 }
 
+void Misc::drawFov() noexcept
+{
+    auto local = localPlayer.get();
```
Comment (Kron1Q) [inline Osiris/Hacks/Misc.cpp:431]: You dont need that line of code, localPlayer, like below.

    if (!localPlayer->isAlive()) return;

Comment (newcommerdontblame) [reply Osiris/Hacks/Misc.cpp:431]: its ok

**Comment by SwagsRGB** [comment]
x @newcommerdontblame can u drop your discord I need something to discuss.

**Comment by newcommerdontblame** [comment]
@SwagsRGB   xEj#5786

File: Osiris/Hacks/Misc.cpp (hunk 1 above)
Comment (caiocinel) [reply Osiris/Hacks/Misc.cpp:441]: It remains inaccurate.

**Comment by Allerek** [comment]
image
after new update

**Comment by caiocinel** [comment]
> image
> after new update
https://github.com/danielkrupinski/Osiris/pull/2218/files#diff-7ab05a48040a01e547b91ccf701fea2b8d76579264bfba3f3819671d71ddb911R431

Before this line add:
`if (!localPlayer || !localPlayer->isAlive())
        return;`

**Comment by flowxrc** [comment]
It isn't drawing anything for me :(

**Comment by memsenpai** [comment]
same
> It isn't drawing anything for me :(

have any config required for it work ?

**Comment by linuxer9** [comment]
Screenshot (31)
here is a better filled circle using Imgui, inspired by onetap
`void Visuals::drawAimbotFOV(ImDrawList* drawList) noexcept
{
    if (!config->visuals.drawAimbotFOV.enabled)
        return;

    if (!localPlayer || !localPlayer->isAlive()) return;

    const auto activeWeapon = localPlayer->getActiveWeapon();
    if (!activeWeapon || !activeWeapon->clip())
        return;

    auto weaponIndex = getWeaponIndex(activeWeapon->itemDefinitionIndex2());
    if (!weaponIndex)
        return;

    auto weaponClass = getWeaponClass(activeWeapon->itemDefinitionIndex2());
    if (!config->aimbot[weaponIndex].enabled)
        weaponIndex = weaponClass;

    if (!config->aimbot[weaponIndex].enabled)
        weaponIndex = 0;
    const auto& screensize = ImGui::GetIO().DisplaySize;
    float radius = std::tan(degreesToRadians(config->aimbot[weaponIndex].fov) / 2.f) / std::tan(degreesToRadians(localPlayer->isScoped() ? localPlayer->fov() : (105 + config->visuals.fov)) / 2.f) * screensize.x;
    ImU32 color = Helpers::calculateColor(config->visuals.drawAimbotFOV.color);
    drawList->AddCircleFilled({ screensize.x / 2.0f, screensize.y / 2.0f }, radius, color, 360);
}`

**Pull request closed without merging**